scripts/dts/gen_defines.py

```diff
@@ -253,23 +253,69 @@ def relativize(path):
 def write_regs(node):
     # Writes address/size output for the registers in the node's 'reg' property
 
-    def write_reg(reg, base_ident, val):
-        # Drop '_0' from the identifier if there's a single register, for
-        # backwards compatibility
-        if len(reg.node.regs) > 1:
-            ident = f"{base_ident}_{reg.node.regs.index(reg)}"
-        else:
-            ident = base_ident
-
-        out_node(node, ident, val,
-                 # Name alias from 'reg-names = ...'
-                 f"{str2ident(reg.name)}_{base_ident}" if reg.name else None)
+    if not node.regs:
+        return
 
-    for reg in node.regs:
-        write_reg(reg, "BASE_ADDRESS", hex(reg.addr))
-        if reg.size:
-            write_reg(reg, "SIZE", reg.size)
+    primary_addrs = {}
+    primary_sizes = {}
```

Comment:
This is a bit confusing to me. Does "primary" mean "path" here too? Noticed that comment above. I'd spell it out in that case.

A nice naming convention is for dicts is `x2y` btw, like `index2addr`.

Reply:
I'm deliberately being vague about which of the node identifiers is the "primary" identifier. Right now it's the `<COMPAT>_<UNIT_ADDR>` style one. Later on it will be the path-based one. Let me see if I can make this clearer. Thanks.